babynames.py

```diff
@@ -42,29 +45,66 @@
 def extract_names(filename):
     """
     Given a single file name for babyXXXX.html, returns a single list starting
-    with the year string followed by the name-rank strings in alphabetical order.
+    with the year string followed by the name-rank strings in alphabetical
+    order.
     ['2006', 'Aaliyah 91', Aaron 57', 'Abagail 895', ' ...]
     """
     names = []
-    # +++your code here+++
-    return names
+
+    names.append(re.findall("\d\d\d\d", filename)[0])
+
+    with open(filename, "r") as f:
+        f = f.read()
+
+    names_list = re.findall("<td>(.+?)</td>", f)
+
+    names_dict = {}
+
+    for index in range(len(names_list)):
+        if index % 3 != 0:
+            if index % 3 == 2:
+                name = names_list[index]
+                value = names_list[index-2]
+            elif index % 3 == 1:
+                name = names_list[index]
+                value = names_list[index-1]
+            if name not in names_dict:
+                names_dict[name] = value
+            elif int(value) < int(names_dict[name]):
+                names_dict[name] = value
+    for name in names_dict:
+        names.append(name + " " + names_dict[name])
+    return sorted(names)
 
 
 def create_parser():
     """Create a cmd line parser object with 2 argument definitions"""
-    parser = argparse.ArgumentParser(description="Extracts and alphabetizes baby names from html.")
+    parser = argparse.ArgumentParser(
+        description="Extracts and alphabetizes baby names from html.")
     parser.add_argument(
         '--summaryfile', help='creates a summary file', action='store_true')
     # The nargs option instructs the parser to expect 1 or more filenames.
-    # It will also expand wildcards just like the shell, e.g. 'baby*.html' will work.
+    # It will also expand wildcards just like the shell, e.g.
+    # 'baby*.html' will work.
     parser.add_argument('files', help='filename(s) to parse', nargs='+')
     return parser
 
 
+def print_info(file_list, create_summary):
```

Comment:
Pulling this out into its own function certainly makes main() a lot cleaner and easier to read.  I'd understood the instructions to say that we HAD to print from main() so I left mine in there.  I hate the way mine looks - so messy there!  This is certainly better coding practice as I understand it!